schematic/schemas/json_schema_validation_rule_functions.py

```diff
@@ -0,0 +1,394 @@
+
+    Returns:
+        A filtered list of validation rules
+    """
+    return [
+        rule
+        for rule in inputted_rules
+        if _get_name_from_inputted_rule(rule) in [e.value for e in ValidationRuleName]
+    ]
+
+
+def check_for_duplicate_inputted_rules(inputted_rules: list[str]) -> None:
+    """Checks that there are no duplicate rules by name
+
+    Arguments:
+        inputted_rules: A list of validation rules
+
+    Raises:
+        ValueError: If there are multiple rules with the same name
+    """
+    rule_names = get_names_from_inputted_rules(inputted_rules)
+    if sorted(rule_names) != sorted(list(set(rule_names))):
+        raise ValueError("Validation Rules contains duplicates: ", inputted_rules)
+
+
+def check_for_conflicting_inputted_rules(inputted_rules: list[str]) -> None:
+    """Checks that each rule has no conflicts with any other rule
+
+    Arguments:
+        inputted_rules: A list of validation rules
+
+    Raises:
+        ValueError: If a rule is in conflict with any other rule
+    """
+    rule_names = get_names_from_inputted_rules(inputted_rules)
+    rules: list[ValidationRule] = _get_rules_by_name(rule_names)
+    for rule in rules:
+        incompatible_rule_names = [rule.value for rule in rule.incompatible_rules]
+        conflicting_rule_names = sorted(
+            list(set(rule_names).intersection(incompatible_rule_names))
+        )
+        if conflicting_rule_names:
+            msg = (
+                f"Validation rule: {rule.name.value} "
+                f"has conflicting rules: {conflicting_rule_names}"
+            )
+            raise ValueError(msg)
+
+
+def get_rule_from_inputted_rules(
+    rule_name: ValidationRuleName, inputted_rules: list[str]
+) -> Optional[str]:
+    """Returns the a rule from a list of rules
+
+    Arguments:
+        rule_name: A ValidationRuleName
+        inputted_rules: A list of validation rules
+
+    Raises:
+        ValueError: If there are multiple of the rule in the list
+
+    Returns:
+        The rule if one is found, otherwise None is returned
+    """
+    inputted_rules = [
+        rule for rule in inputted_rules if rule.startswith(rule_name.value)
+    ]
+    if len(inputted_rules) > 1:
+        raise ValueError("Found duplicates of rule in rules: ", inputted_rules)
+    if len(inputted_rules) == 0:
+        return None
+    return inputted_rules[0]
+
+
+def get_js_type_from_inputted_rules(
+    inputted_rules: list[str],
+) -> Optional[JSONSchemaType]:
+    """Gets the JSON Schema type from a list of rules
+
+    Arguments:
+        inputted_rules: A list of inputted validation rules
+
+    Raises:
+        ValueError: If there are multiple type rules in the list
+
+    Returns:
+        The JSON Schema type if a type rule is found, otherwise None
+    """
+    rule_names = get_names_from_inputted_rules(inputted_rules)
+    validation_rules = _get_rules_by_name(rule_names)
+    json_schema_types = {
```

Comment:
Nit: can we add a comment for this? something like: Collects all unique js_type values from the validation_rules list, skipping any rules where js_type is None